lib/src/controls/flyouts/menu.dart

```diff
@@ -0,0 +1,532 @@
+      );
+    }
+
+    return content;
+  }
+}
+
+// Do not use the platform-specific default scroll configuration.
+// Menus should never overscroll or display an overscroll indicator.
+class _MenuScrollBehavior extends FluentScrollBehavior {
+  const _MenuScrollBehavior();
+
+  @override
+  TargetPlatform getPlatform(BuildContext context) => defaultTargetPlatform;
+
+  @override
+  ScrollPhysics getScrollPhysics(BuildContext context) =>
+      const ClampingScrollPhysics();
+}
+
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+///  * [MenuFlyoutItemBuilder], which renders the given widget in the items list
+abstract class MenuFlyoutItemBase {
+  final Key? key;
+
+  const MenuFlyoutItemBase({this.key});
+
+  Widget build(BuildContext context);
+}
+
+/// Render any widget in the items list of a [MenuFlyout]
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+class MenuFlyoutItemBuilder extends MenuFlyoutItemBase {
+  final WidgetBuilder builder;
+
+  /// Creates a menu flyout item builder
+  const MenuFlyoutItemBuilder({
+    super.key,
+    required this.builder,
+  });
+
+  @override
+  Widget build(BuildContext context) => builder(context);
+}
+
+/// The standart flyout item used inside a [MenuFlyout]
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+class MenuFlyoutItem extends MenuFlyoutItemBase {
+  /// Creates a menu flyout item
+  MenuFlyoutItem({
+    super.key,
+    this.leading,
+    required this.text,
+    this.trailing,
+    required this.onPressed,
+    this.selected = false,
+  });
+
+  /// Displayed before [text].
+  ///
+  /// Don't feel obligated to provide icons for commands that don't have a
+  /// standard visualization. Cryptic icons aren't helpful, create visual
+  /// clutter, and prevent users from focusing on the important menu items.
+  ///
+  /// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/contextmenu_rs2_icons.png)
+  ///
+  /// Usually an [Icon] widget,
+  final Widget? leading;
+  final Widget text;
+  final Widget? trailing;
+  final VoidCallback? onPressed;
+  final bool selected;
```

Comment:
Why not comment these too?

Reply:
I believe it's off the scope of this PR (and I'm lazy)